pkg/utils/utils.go

```diff
@@ -295,3 +300,39 @@ func getPlatform() string {
 	}
 	return PlatformK8s
 }
+
+// GetOCPIngressDomain returns the OpenShift Ingress domain
+func GetOCPIngressDomain() (string, error) {
```

Comment:
This call to be moved level up, to controller.
The simplest option I can see is to extend ExternalConfig with ingress domain field  and make it filled by spec-preprocessor (same as other ExternalConfig fields)
This way it is possible to add/move the testing of logic to unit tests instead of integration tests.